### DISCUSSION TITLE
Database as a Kubernetes deployment or a cloud service?

### DISCUSSION THREAD

**Discussion by sergiorivas** [BretFisher/ama#168] [Q&A] [answered]
I wonder what's better?

1. Use the cloud provide database service (For example RDS in AWS) and link my pods to that service
2. Create a deployment into our Kubernetes cluster (For example, having PostgreSQL pods with volume)

Thanks.

**Comment by BretFisher** [comment, marked as answer]
I always recommend using a cloud-provider database or database SaaS. RDS has lots of functionality built-in that you would have to spend a lot of time replicating in your own cluster (failover, backups, scale up, restores, monitoring, upgrades, storage management, admin permissions, RBAC, etc.).

Compared to traditional server setup, Kubernetes makes databases easier to deploy and update, and with Operators, it may be easier to make highly available and backed up than before, but it's still a lot of work to take on if you don't have clear business reasons to do so.

**Comment by FinalDes** [comment]
what is the different to use Operator vs statefullset @BretFisher ?

**Comment by BretFisher** [comment]
They are not related. Operators are automation that add functionality to Kubernetes. StatefullSet is one of the ways to deploy containers.
